bundle/src/main/java/com/adobe/acs/commons/mcp/impl/processes/ReplicatorQueue.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017 Adobe.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.adobe.acs.commons.mcp.impl.processes;
+
+import com.day.cq.replication.ReplicationAction;
+import com.day.cq.replication.ReplicationActionType;
+import com.day.cq.replication.ReplicationContentFilter;
+import com.day.cq.replication.ReplicationException;
+import com.day.cq.replication.ReplicationOptions;
+import com.day.cq.replication.ReplicationStatus;
+import com.day.cq.replication.Replicator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import javax.jcr.Session;
+
+/**
+ * Collect a list of replication events for later examination and/or replay.
+ */
+public class ReplicatorQueue implements Replicator {
+    Map<String, ReplicationOptions> deactivateOperations = new LinkedHashMap<>();
```

Comment:
Is this class's state thread safe? It looks like any PageRelocator processes will be reading/writing from this same object in an unsynchronized manner.